bb8/src/inner.rs

```diff
@@ -127,7 +134,16 @@ where
             }
         };
 
-        self.inner.statistics.record(kind);
+        self.inner.statistics.record_get(kind);
+
+        if let Some(wait_time_start) = wait_time_start {
+            let wait_time = Instant::now() - wait_time_start;
+            self.inner
+                .statistics
+                .get_waited_time_micro
+                .fetch_add(wait_time.as_micros() as u64, Ordering::SeqCst);
```

Comment:
Why micros vs millis or nanos?

Reply:
I guess that micros would provide enough detail (nanos might be just noise) while we would keep the also the u64 large enough for storing many many many micros.

In any case since we move this to a Duration we will be storing behind the scenes IIRC micros using u128.

Reply:
Oks now I remembered, everything I wanted to use an Atomic Type. I guess that we can try to use an `u128` internally and return a Duration (which IIRC is just a u128). If this works we could just forget about the discussion between nanos and micros.

Reply:
Finally I went for using micros with u64 for converting later to Duration. If you want to use nanos, i can do that. But got the feeling that nanosecond resolution for this specific use case would not provide a lot of value (indeed micro most likely will not neither)

Ive just kept the type to `u64` just for having all atomics using the same type and helping me to keep the `record` interface simple.

If you want to to go for nanos and/or `u128` let me know.